doc/HOWTO-DAEMON.md

```diff
@@ -0,0 +1,159 @@
+(and the restarted config file may name a different lockdir).
+
+### `[sample]` section aka `[ps]` section
+
+```
+cadence = <duration value>
+exclude-system-jobs = <bool>                    # default true
+load = <bool>                                   # default true
+batchless = <bool>                              # default false
+exclude-users = <comma-separated strings>       # default []
+exclude-commands = <comma-separated strings>    # default []
+```
+
+These are the normal options for `sonar ps`, see the Sonar documentation.
+
+### `[sysinfo]` section
+
+```
+cadence = <duration value>
+on-startup = <bool>                             # default true
+```
+
+If `on-startup` is `true` then a sysinfo operation will be executed every time the daemon is
+started, in addition to according to the cadence.
+
+### `[jobs]` section aka `[slurm]` section
+
+```
+cadence = <duration value>
+window = <duration value>                       # default 2*cadence
+uncompleted = <bool>                            # default false
+delta-coding = <bool>                           # default true
+dump = <filename>
+```
+
+The window is the sacct time window used for looking for data.
+
+The `uncompleted` option, if true, triggers the inclusion of data about pending and running jobs.
+This will result in multiple transmissions of data for the same `(job_id,job_step)`, one at each
+sample point.  If a job stays in, say, the PENDING state for several sampling windows then multiple
+transmissions for the job in the PENDING state will be seen.
+
+The `delta-coding` option, if true, triggers optimization of data transmission: redundant data are
+omitted in subsequent transmissions, and if there are no pertient data changes - for example, if a
+new `PENDING` record has the same contents as one already sent because no settings have changed -
+then the second record will not be sent at all.  What is deemed pertient or redundant is up for
+discussion for both `PENDING` and `RUNNING` jobs.  In any case, the recipient must be prepared to
+reconstruct the complete data stream from the initial record for the `(job_id,job_step)` and any
+subsequent deltas, applied in order.  If `uncompleted` is true, the `delta-coding` option can
+greatly reduce transmitted data volume.  If `uncompleted` is false, it can still avoid redundantly
+sending information about completed jobs.
+
+Setting `dump` to a file name will cause JSON output to be appended to the given file, while also
+being sent in the normal way.  If appending fails, a diagnostic is printed if `[debug]:verbose` is
+true.
+
+### `[cluster]` section
+
+```
+cadence = <duration value>
+```
+
+### `[debug]` section
+
+```
+dump = bool                                     # default false
+verbose = bool                                  # default false
+```
+
+Setting `dump` to true will cause the daemon to dump all data it is sending on stdout.
```

Comment:
Not obvious that a dump to stdout is meaningful when stdio is the only i/o channel.  We could dump to stderr.  We could (re)move the function, which is really for the benefit of Kafka, and add it back for Kafka, but then in the Kafka section?

If we remove dump here and add it back elsewhere, possibly `verbose` should move to the global section.

Reply:
I'm moving dump to the kafka code and keeping verbose here.